### DISCUSSION TITLE
WritableBitmap image stretch issue

### DISCUSSION THREAD

**Discussion by sbartolett3** [unoplatform/uno#11923] [Q&A] [answered]
Hi,

I'm trying to use a WriteableBitmap to stream a camera image to an Image control.  Running on Windows, everything is working as expected.  However when I run the same code on Linux, if I set the image controls stretch property to Uniform, the image is first displayed with no stretch and then redraw with the required stretch.  This causes the displayed image to flash.  If I set the stretch property to None, then the image displays ok.  For my application it will be required to have stretch set to Uniform.

The .NET implementation of WriteableBitmap has lock/unlock methods.  These don't seem to be supported in Uno.  Not sure if this would help or not.

Here's the relevant code:

WriteableBitmap singleFrame = null;
ImageSource myImage = null;
myImage = singleFrame;
.
.
.
                dispatcherQueue.TryEnqueue(() =>
                {
                    using (Stream stream = singleFrame.PixelBuffer.AsStream())
                    {
                        stream.Write(imageBufferRGB, 0, imageBufferRGB.Length);
                    }
                    if (myImage != null)
                        singleFrame.Invalidate();
                });

**Comment by jeromelaban** [comment]
Thanks for the report. Can you open an issue and provide a repro sample of what you trying to achieve, so we're sure to talk about the same issue?

**Reply by sbartolett3** [reply, marked as answer]
Issue #11930 just opened with sample app to demonstrate problem.
